polaris/_mixins.py

```diff
@@ -0,0 +1,70 @@
+import abc
+
+from loguru import logger
+from pydantic import BaseModel, PrivateAttr, computed_field
+
+from polaris.utils.errors import PolarisChecksumError
+
+
+class ChecksumMixin(BaseModel, abc.ABC):
+    """
+    Mixin class to add checksum functionality to a class.
+    """
+
+    _md5sum: str | None = PrivateAttr(None)
+
+    @abc.abstractmethod
+    def _compute_checksum(self) -> str:
+        """Compute the checksum and return it along with manifest."""
+        raise NotImplementedError
+
+    @computed_field
+    @property
+    def md5sum(self) -> str:
+        """Lazily compute the checksum once needed."""
+        if not self.has_md5sum:
+            logger.info("Computing the checksum. This can be slow for large datasets.")
+            self.md5sum = self._compute_checksum()
+        return self._md5sum
+
+    @md5sum.setter
+    def md5sum(self, value: str):
+        """Set the checksum."""
+        if len(value) != 32 or not all(c in "0123456789abcdef" for c in value):
```

Comment:
**Suggestion**
An instance where a regular expression is your friend:
```suggestion
        if not re.fullmatch(r"^[a-f0-9]{32}$", value):
```

Reply:
I want to get more comfortable with REs. It's always been a little daunting to me!